### PR TITLE
Support for optional as UnionType (Python >= 3.10)

### PR DESCRIPTION
**Pull request by mkomet**
This resolves the issue https://github.com/tiangolo/typer/issues/533

### CODE REVIEW

**Comment by johnthagen** [comment]
Also closes 

- #348

Another PR that also address this issue

- #676

File: typer/main.py [hunk 1]
```diff
@@ -816,8 +822,9 @@ def get_click_param(
     is_tuple = False
     parameter_type: Any = None
     is_flag = None
+    is_union_type = lenient_issubclass(type(main_type), UnionType)
```
Comment (duongdominhchau) [inline typer/main.py:825]: Hi, I'm new to this kind of stuff, could you please clarify this for me please? From the variable name, it is a `type` already, why do we need to convert it to `type()` here? Is it just your preference or there are some edge cases being handled implicitly here?

**Review by svlandeg — COMMENTED** [review]
Hi, thanks for this PR!

There's a few other PRs that address the same issue, of which I think #548 would be the prime candidate to merge. That PR uses existing functionality such as `utils.needs_py310` and `._typing.is_union`.  If we merge #548, I think we can go ahead and close this one.

I'll leave the final decision with Tiangolo though.

**Comment by tiangolo** [comment]
Thank you for the work @mkomet! :coffee: 

And thanks @svlandeg for the help :bow: 

This should have been fixed by https://github.com/fastapi/typer/pull/548

It is now available in Typer `0.12.4` :rocket: 

Given that, I'll close this one.

**Pull request closed without merging**